addon/components/polaris-text-field.js

```diff
@@ -623,13 +622,13 @@ export default class PolarisTextField extends Component {
 
   @action
   handleKeyPress(event) {
-    let { key } = event;
+    let { key, which } = event;
     let numbersSpec = /[\d.eE+-]$/;
 
     if (
       this.type !== 'number' ||
-      getCode(event) === 'Enter' ||
-      key.match(numbersSpec)
+      which === 13 /* Enter key */ ||
+      numbersSpec.test(key)
```

Comment:
These changes are based on the [React implementation](https://github.com/Shopify/polaris-react/blob/master/src/components/TextField/TextField.tsx#L479).